qiskit/qpy/interface.py

```diff
@@ -12,16 +12,30 @@
 
 """User interface of qpy serializer."""
 
+from json import JSONEncoder, JSONDecoder
+from typing import Union, List, BinaryIO
+from collections.abc import Iterable
 import struct
 import warnings
 
 from qiskit.circuit import QuantumCircuit
+from qiskit.pulse import ScheduleBlock
 from qiskit.exceptions import QiskitError
-from qiskit.qpy import formats, common, binary_io
+from qiskit.qpy import formats, common, binary_io, type_keys
+from qiskit.qpy.exceptions import QpyError
 from qiskit.version import __version__
 
 
-def dump(circuits, file_obj, metadata_serializer=None):
+# pylint: disable=invalid-name
+QPY_SUPPORTED_TYPES = Union[QuantumCircuit, ScheduleBlock]
+
+
+def dump(
+    programs: Union[List[QPY_SUPPORTED_TYPES], QPY_SUPPORTED_TYPES],
+    file_obj: BinaryIO,
+    metadata_serializer: JSONEncoder = None,
+    circuits: QuantumCircuit = None,
```

Comment:
Isn't there a decorator that does this implicitly for us?

Reply:
Thanks, I had forgotten about it :)